dynosaur_derive/src/lib.rs

```diff
@@ -235,11 +235,19 @@ fn mk_dyn_struct_impl_item(struct_ident: &Ident, item_trait: &ItemTrait) -> Toke
                  sig,
                  ..
              }| {
-                 let args = invoke_fn_args(sig, false);
                  let ident = &sig.ident;
+                 let mut sig = sig.clone();
+                 let args = invoke_fn_args(&sig, false);
+
+                 let (ret_arrow, ret) = expand_async_ret_ty(&sig);
+                 sig.output = parse_quote! { #ret_arrow impl #ret  };
+                 remove_asyncness_from_fn(&mut sig);
+
                  quote! {
                      #sig {
-                         unsafe { &*self.ptr }.#ident(#(#args),*).await
+                         let fut: ::core::pin::Pin<Box<dyn #ret + '_>> = unsafe { &*self.ptr }.#ident(#(#args),*);
+                         let fut: ::core::pin::Pin<Box<dyn #ret + '_>> = unsafe { ::core::mem::transmute(fut) };
```

Comment:
I have no idea what lifetime it could be inferring here for the second `'_`. I'm surprised that it compiles when there are multiple possibilities. Seems brittle since inference behavior could change in the future.. I'm not sure it will continue to compile.